app/src/ui/stashing/stash-diff-header.tsx

```diff
@@ -117,8 +92,13 @@ export class StashDiffHeader extends React.Component<
   private onRestoreClick = async () => {
     const { dispatcher, repository, stashEntry } = this.props
 
-    this.setState({ isRestoring: true }, () => {
-      dispatcher.popStash(repository, stashEntry)
-    })
+    try {
+      this.setState({ isRestoring: true })
+      await dispatcher.popStash(repository, stashEntry)
+    } catch (err) {
+      dispatcher.postError(err)
```

Comment:
Just for completeness, please wrap this `err` in an `ErrorWithMetadata` instance:

```suggestion
      const errorWithMetadata = new ErrorWithMetadata(err, {
        repository: repository,
      })
      dispatcher.postError(errorWithMetadata)
```

**Beware you will need to import `ErrorWithMetadata` at the top of the file:**
![image](https://github.com/desktop/desktop/assets/1083228/e3b279d4-11d5-46e8-baef-dd9e64632e69)

Reply:
Thanks for letting me know.
I have implemented the changes.

(btw, it's my birthday. Thanks for looking at this PR.)